Automattic/newspack-plugin
# Performance Wizard

### All Submissions:






### Changes proposed in this Pull Request:



This PR introduces the Performance Wizard. This wizard names [Progressive WP](https://wordpress.org/plugins/progressive-wp/) and [PWA](https://wordpress.org/plugins/pwa/) as dependencies, and configures Progressive WP for Save To Homescreen, Offline Usage, and Push Notification features. In addition to normal Wizard activity, this adds a few other things:

- `withWizardScreen` has a `noCard` option. In this option the screen is not wrapped in a `Card` which allows for multi-Card layouts like the first screen of the wizard.
- The configuration manager updates another plugin's options, that are serialized in a single option.
- The configuration manager gets and updates one field which is part of core—the Site Icon.

Note: Newspack does not yet have a Muriel-ized Toggle control, so I'm using the default WordPress one. I'll file an issue and we can create one shortly and swap it in. This is why the Tooltips from @sonjaleix's designs are missing. 

### How to test the changes in this Pull Request:

1. `npm ci && npm run clean && npm run build:webpack`
2. Navigate to the Performance Wizard.
3. Verify PWA and Progressive WP plugins are installed and activated successfully.
4. On the `Add To Homescreen` screen, verify that the Site Icon you see is the one uploaded in the Customizer. 
5. Replace the Site Icon and verify that the change persists to the Customizer as well as this Wizard.
6. On the `Push Notifications` screen, update Server Key and Sender ID fields. 
7. Verify that these fields are persisted.
8. Go to the Progressive WP dashboard, then the Push Notifications section, and verify that the values you input are present there as well.

### Other information:
